src/processes.rs

```diff
@@ -0,0 +1,183 @@
+/// use std::io::Write;
+/// use libproc::processes;
+///
+/// if let Ok(pids) = processes::pids_by_path("/etc/hosts", false, false) {
+///     println!("Found {} processes accessing /etc/hosts", pids.len());
+/// }
+/// ```
+///
+/// List all processes that have a file open on a specific volume; the path
+/// argument is used to get the filesystem device ID, and processes that have
+/// a file open on that same device ID match the filter:
+/// ```
+/// use std::io::Write;
+/// use libproc::processes;
+///
+/// if let Ok(pids) = processes::pids_by_path("/Volumes/MountedDrive", true, false) {
+///     println!("Found {} processes accessing files on /Volumes/MountedDrive", pids.len());
+/// }
+/// ```
+#[cfg(target_os = "macos")]
+pub fn pids_by_path(
+    path: &str,
+    is_volume: bool,
+    exclude_event_only: bool,
+) -> Result<Vec<u32>, String> {
+    let path_flags = is_volume as u32 | exclude_event_only as u32;
+    listpidspath(ProcFilter::All, path, path_flags)
+}
+
+/// Returns a filtered list of PIDs of active processes that reference reference
+/// an open file with the given path or volume, with or without files opened
+/// with the `O_EVTONLY` flag. Use a [`ProcFilter`] member to specify how to
+/// filter the list of PIDs returned.
+///
+/// (Files opened with the `O_EVTONLY` flag will not prevent a volume from being
+/// unmounted).
+///
+/// # Examples
+///
+/// Get the list of process ids for child processes that have a specific filename open:
+///
+/// ```
+/// use std::io::Write;
+/// use libproc::libproc::processes;
+///
+/// let filter = ProcFilter::ByParentProcess { ppid: process::id() };
+/// if let Ok(pids) = processes::pids_by_type_and_path(filter, "/etc/hosts", false, false) {
+///     println!("Found {} processes accessing /etc/hosts", pids.len());
+/// }
+/// ```
+///
+/// List all processes within the current process group that have a file open on
+/// a specific volume; the path argument is used to get the filesystem device
+/// ID, and processes that have a file open on that same device ID match the
+/// filter:
+/// ```
+/// use std::io::Write;
+/// use libproc::libproc::processes;
+///
+/// let filter = ProcFilter::ByParentProcess { ppid: process::id() };
+/// if let Ok(pids) = processes::pids_by_type_and_path("/Volumes/MountedDrive", true, false) {
+///     println!("Found {} processes accessing files on /Volumes/MountedDrive", pids.len());
+/// }
+/// ```
+#[cfg(target_os = "macos")]
+pub fn pids_by_type_and_path(
+    filter: ProcFilter,
+    path: &str,
+    is_volume: bool,
+    exclude_event_only: bool,
+) -> Result<Vec<u32>, String> {
+    let path_flags = is_volume as u32 | exclude_event_only as u32;
+    listpidspath(filter, path, path_flags)
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn listpids_test() {
+        let pids = pids_by_type(ProcFilter::All).expect("Could not list pids");
+        assert!(pids.len() > 1);
+    }
+
+    #[test]
+    #[cfg(target_os = "macos")]
```

Comment:
You'd remove all the "macos" lines when you finish the linux part, and have processes.rs be cross-platform (tests and all)?

Reply:
Not this part, unfortunately. It's a lot more work to reimplement all the work the Darwin libproc library does here.